### PR TITLE
doc: add notes on benchmarks

### PR DESCRIPTION
**Pull request by Alizter**
We add some notes on the various types of benchmarks we have available.

### CODE REVIEW

File: doc/hacking.rst [hunk 1]
```diff
@@ -527,3 +527,36 @@ Good:
...
    let (foo : _ Command.Args.t) = S []
+
+Benchmarking
+============
+
+Dune Bench
+----------
+
+You can benchmark Dune's performance by running `make bench`. This will run a
+subset of the duniverse.
```
Comment (christinerose) [inline doc/hacking.rst:538]:
```suggestion
subset of the Duniverse.
```

does ‘make bench’ need two ‘ on either side to make it monospace in .rst files?

File: doc/hacking.rst [hunk 2]
```diff
@@ -527,3 +527,36 @@ Good:
...
+
+Inline Benchmarks
+-----------------
+
+Certain performance critical parts of Dune are benchmarked using the
```
Comment (christinerose) [inline doc/hacking.rst:543]:
```suggestion
Certain performance-critical parts of Dune are benchmarked using the
```

File: doc/hacking.rst [hunk 3]
```diff
@@ -527,3 +527,36 @@ Good:
...
+``inline_benchmarks`` library. These benchmarks are run when running the tests.
+Their outputs are currently not recorded and are only used to detect performance
+regressions.
+
+
+Build-time Benchmarks
```
Comment (christinerose) [inline doc/hacking.rst:549]:
```suggestion
Build-Time Benchmarks
```

**Review by christinerose — APPROVED** [review]
Minor formatting / style suggestions.

File: doc/hacking.rst (hunk 1 above)
Comment (Alizter) [reply doc/hacking.rst:538]: That was supposed to be a backtick my bad.

**Pull request merged**